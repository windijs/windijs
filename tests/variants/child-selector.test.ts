import { all, backgroundClipConfig, backgroundColor, bundle, children, colors, configHandler, createUtility, prop, sibling, siblings, svg } from "index";

const bg = createUtility("bg")
  .use(backgroundColor(colors))
  .case("clip", configHandler(backgroundClipConfig, ["backgroundClip", prop`-webkit-background-clip`]))
  .init();

const utilities = [bg.blue[500], bg.clip.content];

test("svg", () => {
  expect(bundle(svg(...utilities))).toMatchSnapshot();
});

test("all", () => {
  expect(bundle(all(...utilities))).toMatchSnapshot();
});

test("children", () => {
  expect(bundle(children(...utilities))).toMatchSnapshot();
});

test("siblings", () => {
  expect(bundle(siblings(...utilities))).toMatchSnapshot();
});

test("sibling", () => {
  expect(bundle(sibling(...utilities))).toMatchSnapshot();
});
