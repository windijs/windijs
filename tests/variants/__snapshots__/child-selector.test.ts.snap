// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`all 1`] = `
Object {
  "& *": Object {
    "--w-bg-opacity": "1",
    "-webkit-background-clip": "content-box",
    "backgroundClip": "content-box",
    "backgroundColor": "rgba(59, 130, 246, var(--w-bg-opacity))",
  },
}
`;

exports[`children 1`] = `
Object {
  "& > *": Object {
    "--w-bg-opacity": "1",
    "-webkit-background-clip": "content-box",
    "backgroundClip": "content-box",
    "backgroundColor": "rgba(59, 130, 246, var(--w-bg-opacity))",
  },
}
`;

exports[`sibling 1`] = `
Object {
  "& + *": Object {
    "--w-bg-opacity": "1",
    "-webkit-background-clip": "content-box",
    "backgroundClip": "content-box",
    "backgroundColor": "rgba(59, 130, 246, var(--w-bg-opacity))",
  },
}
`;

exports[`siblings 1`] = `
Object {
  "& ~ *": Object {
    "--w-bg-opacity": "1",
    "-webkit-background-clip": "content-box",
    "backgroundClip": "content-box",
    "backgroundColor": "rgba(59, 130, 246, var(--w-bg-opacity))",
  },
}
`;

exports[`svg 1`] = `
Object {
  "& svg": Object {
    "--w-bg-opacity": "1",
    "-webkit-background-clip": "content-box",
    "backgroundClip": "content-box",
    "backgroundColor": "rgba(59, 130, 246, var(--w-bg-opacity))",
  },
}
`;
